ledger/block/src/transaction/merkle.rs

```diff
@@ -173,6 +173,20 @@ impl<N: Network> Transaction<N> {
         N::merkle_tree_bhp::<TRANSACTION_DEPTH>(&leaves)
     }
 
+    /// Returns the Merkle tree for the given execution tree and fee.
+    pub fn transaction_tree(
+        mut execution_tree: TransactionTree<N>,
+        fee_index: usize,
+        fee: &Fee<N>,
+    ) -> Result<TransactionTree<N>> {
+        // Construct the transaction leaf.
+        let leaf = TransactionLeaf::new_fee(u16::try_from(fee_index)?, **fee.transition_id()).to_bits_le();
+        // Compute the transaction tree.
+        execution_tree.append(&[leaf])?;
+
+        Ok(execution_tree)
+    }
+
```

Comment:
Just for cleanliness, I'm not sure I like this particular approach because there is technically 2 paths to get a transaction tree now:

1. Creating a tree directly with a deployment/execution + fee
2. Creating a deployment/execution tree and then concatenating the fee

Is there a way to unify these (where one relies on the other)? Also for safety, `transaction_tree` should not be public.

Reply:
> Is there a way to unify these (where one relies on the other) into a single function?

Good one, see: 1f2b23e7e

Also added explicit types for the intermediary trees, to avoid future developers mistaking the outputs: b05b4f151

I'm on the fence whether new DeploymentID and ExecutionID types (similar to TransactionID) would be a net benefit. This would avoid accidental misinterpretation, but in constrat to TransactionID, we don't "need" those new types in most of the codebase.

> Also for safety, transaction_tree can be made private.

Doesn't work unfortunately, because its not a "method" taking `&self`.